Using ion-js directly from git repository does not work

Currently when adding ion-js as a dependency in npm using direct git syntax (either direct Git url, or direct GitHub url), the resulting library does not work. The resulting folder does not contain the source at all.

I'm not really sure what all is causing this, but there should probably at least be a NPM `prepare` script which will do the compilation of the source code in to a usable library by running `grunt build` or `grunt release`:

> prepare: Run both BEFORE the package is packed and published, on local npm install without any arguments, and when installing git dependencies (See below).

Sure. The key is `npm install amzn/ion-js#master`

Here is a full repro:
```
naked@ovo:~/lskdjf$ mkdir repro
naked@ovo:~/lskdjf$ cd repro
naked@ovo:~/lskdjf/repro$ npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help json` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg>` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
package name: (repro) 
version: (1.0.0) 
description: 
entry point: (index.js) 
test command: 
git repository: 
keywords: 
author: 
license: (ISC) 
About to write to /home/naked/lskdjf/repro/package.json:

{
  "name": "repro",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}


Is this OK? (yes) 
naked@ovo:~/lskdjf/repro$ npm install amzn/ion-js#master
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN repro@1.0.0 No description
npm WARN repro@1.0.0 No repository field.

+ ion-js@3.1.2
added 2 packages from 1 contributor and audited 2 packages in 5.808s
found 0 vulnerabilities



   ╭────────────────────────────────────────────────────────────────╮
   │                                                                │
   │       New minor version of npm available! 6.6.0 → 6.13.0       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v6.13.0   │
   │               Run npm install -g npm to update!                │
   │                                                                │
   ╰────────────────────────────────────────────────────────────────╯

naked@ovo:~/lskdjf/repro$ ls
node_modules  package.json  package-lock.json
naked@ovo:~/lskdjf/repro$ node
> ion = require('ion-js')
Thrown:
{ Error: Cannot find module 'ion-js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18) code: 'MODULE_NOT_FOUND' }
> naked@ovo:~/lskdjf/repro$ find node_modules/ion-js/
node_modules/ion-js/
node_modules/ion-js/RELEASE.md
node_modules/ion-js/tsconfig.commonjs.json
node_modules/ion-js/package.json
node_modules/ion-js/typedoc.json
node_modules/ion-js/LICENSE
node_modules/ion-js/tsconfig.amd.json
node_modules/ion-js/test
node_modules/ion-js/test/IonText.ts
node_modules/ion-js/test/IonImport.ts
node_modules/ion-js/test/textNulls.ts
node_modules/ion-js/test/AbstractWriter.ts
node_modules/ion-js/test/IonBinaryTimestamp.ts
node_modules/ion-js/test/writeSymbolTokens.ts
node_modules/ion-js/test/iontests.ts
node_modules/ion-js/test/IonTextReader.ts
node_modules/ion-js/test/mocha.opts
node_modules/ion-js/test/spans.ts
node_modules/ion-js/test/IonBinaryReader.ts
node_modules/ion-js/test/IntSize.ts
node_modules/ion-js/test/IonTimestamp.ts
node_modules/ion-js/test/IonLowLevelBinaryWriter.ts
node_modules/ion-js/test/IonLocalSymbolTable.ts
node_modules/ion-js/test/IonUnicode.ts
node_modules/ion-js/test/IonCatalog.ts
node_modules/ion-js/test/IonTextWriter.ts
node_modules/ion-js/test/IonBinaryWriter.ts
node_modules/ion-js/test/IonDecimal.ts
node_modules/ion-js/test/IonParserBinaryRaw.ts
node_modules/ion-js/test/IonAnnotations.ts
node_modules/ion-js/test/util.ts
node_modules/ion-js/test/IonWriteable.ts
node_modules/ion-js/CONTRIBUTE.md
node_modules/ion-js/README.md
node_modules/ion-js/NOTICE
node_modules/ion-js/tsconfig.json
```

That is, ion-js does not work when using the repository directly, instead of a release tarball. Same thing for manually checking out the full repository and using `npm link`.

--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
   "types": "dist/commonjs/es6/Ion.d.ts",
   "scripts": {
     "commit": "git-cz",
+    "prepare": "grunt",
     "test": "nyc mocha",
     "release": "grunt release"
   },

